packages/core/src/browser/core-preferences.ts

```diff
@@ -78,6 +79,8 @@ export function createCorePreferences(preferences: PreferenceService): CorePrefe
 }
 
 export function bindCorePreferences(bind: interfaces.Bind): void {
+    corePreferenceSchema.properties['workbench.colorTheme'].default = FrontendApplicationConfigProvider.get().defaultTheme || 'dark';
```

Comment:
you should be able to inline it in the preference schema definition, `FrontendApplicationConfigProvider.get()` should be available always

Reply:
I have tried to make these two `defaults` inline but then the browser will show the error:
```
Uncaught Error: The configuration is not set. Did you call FrontendApplicationConfigProvider#set?
```
this error should be related to the order of https://github.com/eclipse-theia/theia/blob/6a6a673848c57a1776da66a063d38072768e3ee3/dev-packages/application-manager/src/generator/frontend-generator.ts#L73 and https://github.com/eclipse-theia/theia/blob/6a6a673848c57a1776da66a063d38072768e3ee3/dev-packages/application-manager/src/generator/frontend-generator.ts#L82 in  `frontend-generator.ts`

so I'm going to move `FrontendApplicationConfigProvider.set` to the top of the generated `index.js`.

Reply:
yes, it should be fine

Reply:
now the unit test is failed for the similar error, so maybe putting these two `default`s in `bindCorePreferences` is a better choice?

Reply:
or we expose `FrontendApplicationConfigProvider.doGet` for such cases

Reply:
good advice

Reply:
pass the unit tests